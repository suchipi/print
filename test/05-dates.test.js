const print = require("../print");

describe("Dates", () => {
	it("prints dates", () => {
		const date = "2000-12-31T18:02:16.555Z";
		expect(print(new Date(date))).toMatchSnapshot();
	});

	it("prints properties", () => {
		const date = new Date("2000-10-10T10:02:02.000Z");
		date.foo = "bar";
		date.list = ["Alpha", "Beta", "Delta"];
		expect(print(date)).toMatchSnapshot();
	});

	it("identifies malformed dates", () => {
		const date = new Date(NaN);
		expect(print(date)).toMatchSnapshot();
		date.foo = "bar";
		expect(print(date)).toMatchSnapshot();
	});

	it("identifies subclasses", () => {
		class Timestamp extends Date {}
		const date = "2000-12-31T18:02:16.555Z";
		expect(print(new Timestamp(date))).toMatchSnapshot();

		class BadDate extends Date {
			constructor() {
				super(NaN);
			}
		}
		const bad8 = new BadDate();
		expect(print(bad8)).toMatchSnapshot();
		const value = "Yeah, that's bad";
		Object.defineProperty(bad8, "text", { value });
		expect(print(bad8)).toMatchSnapshot();
		expect(print(bad8, { all: true })).toMatchSnapshot();
	});
});
