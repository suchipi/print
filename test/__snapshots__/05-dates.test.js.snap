// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dates identifies malformed dates 1`] = `Invalid Date`;

exports[`Dates identifies malformed dates 2`] = `
Date {
	Invalid Date
	
	foo: "bar"
}
`;

exports[`Dates identifies subclasses 1`] = `
Timestamp {
	2000-12-31T18:02:16.555Z
}
`;

exports[`Dates identifies subclasses 2`] = `
BadDate {
	Invalid Date
}
`;

exports[`Dates identifies subclasses 3`] = `
BadDate {
	Invalid Date
}
`;

exports[`Dates identifies subclasses 4`] = `
BadDate {
	Invalid Date
	
	text: "Yeah, that's bad"
}
`;

exports[`Dates prints dates 1`] = `2000-12-31T18:02:16.555Z`;

exports[`Dates prints properties 1`] = `
Date {
	2000-10-10T10:02:02.000Z
	
	foo: "bar"
	list: [
		"Alpha"
		"Beta"
		"Delta"
	]
}
`;
